config/src/main/java/com/typesafe/config/impl/Path.java

```diff
@@ -141,6 +154,21 @@ Path subPath(int firstIndex, int lastIndex) {
         return pb.result();
     }
 
+    boolean startsWith(Path other) {
+        Path myRemainder = this;
+        Path otherRemainder = other;
+        if (otherRemainder.length() <= myRemainder.length()) {
```

Comment:
since length is O(n) it would be a bit nicer to instead check myRemainder for null, below, perhaps. Since paths are generally short it probably doesn't really matter that much.